ambry-router/src/test/java/com/github/ambry/router/CompressionServiceTest.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.compression.CompressionException;
+import com.github.ambry.compression.LZ4Compression;
+import com.github.ambry.compression.ZstdCompression;
+import com.github.ambry.config.CompressionConfig;
+import com.github.ambry.messageformat.BlobProperties;
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.TestUtils;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.nio.ByteBuffer;
+import org.apache.commons.lang3.tuple.Triple;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class CompressionServiceTest {
+
+  @Test
+  public void testIsBlobCompressible() {
+    BlobProperties blobProperties = new BlobProperties(1234L, "testServiceID", (short) 2222, (short) 3333, true);
+    CompressionMetrics metrics = new CompressionMetrics(new MetricRegistry());
+    CompressionConfig config = new CompressionConfig();
+    CompressionService service = new CompressionService(config, metrics);
+
+    // Test: compression disabled.  Compression does nothing.
+    config.isCompressionEnabled = false;
```

Comment:
hmm, configuration's members are not read-only. i would rather make it read-only and create a new configuration from a Properties.

Reply:
Whether to make it read-only or not really depends on what "config" is.  If we use "config" as for a one-time remote-control purpose, we can make it read-only and feel free to make as many copies as wanted.  Any changes will require restarting the service that is very expensive.  If the purpose of config is to control run-time behavior of a service, then config should be read-write and singleton.  Admin should be able to change run-time behavior without having to restart the service.  I think configuration should be run-time behavior controller that can alter at run-time without restarting services and the service should not cache config values.